torch/_inductor/codegen/cpp_bmm_template.py

```diff
@@ -0,0 +1,285 @@
+                L.view(blocked_w, (b, k, padded_n // block_n, block_n)),
+                [0, 2, 1, 3],
+            )
+            assert micro_gemm.get_b_layout() != LayoutType.NORMAL
+            vnni_size = 4 if micro_gemm.get_b_layout() == LayoutType.VNNI4 else 2
+            blocked_w = L.view(
+                L.permute(
+                    L.view(
+                        blocked_w,
+                        (b, padded_n // block_n, k // vnni_size, vnni_size, block_n),
+                    ),
+                    [0, 1, 2, 4, 3],
+                ),
+                [b] + new_size,
+            )
+            blocked_w = ir.ExternKernel.realize_input(blocked_w)
+            blocked_w = ir.ExternKernel.require_contiguous(blocked_w)
+            if isinstance(blocked_w, ir.ReinterpretView):
+                # normalize stride to be "contiguous_strides" per size
+                # this avoids the problems in L.view during template codegen
+                assert isinstance(blocked_w.layout, ir.FixedLayout)
+                blocked_w.layout = ir.FixedLayout(
+                    blocked_w.layout.device,
+                    blocked_w.layout.dtype,
+                    blocked_w.layout.size,
+                    ir.FlexibleLayout.contiguous_strides(blocked_w.layout.size),
+                    blocked_w.layout.offset,
+                )
+        else:
+            blocked_w = (
+                torch.nn.functional.pad(W, (0, padded_n - n))
+                .reshape(-1, k, padded_n // block_n, block_n)
+                .transpose(1, 2)
+                .contiguous()
+            )
+            assert micro_gemm.get_b_layout() != LayoutType.NORMAL
+            layout_str = (
+                "VNNI4" if micro_gemm.get_b_layout() == LayoutType.VNNI4 else "VNNI2"
+            )
+            assert micro_gemm.get_b_layout() in [
+                LayoutType.VNNI2,
+                LayoutType.VNNI4,
+            ], f"We only support {layout_str} for now"
+            vnni_size = 4 if micro_gemm.get_b_layout() == LayoutType.VNNI4 else 2
+            assert (
+                k % vnni_size == 0
+            ), f"k should be divisible by vnni_size for {layout_str} layout"
+            blocked_w = (
+                blocked_w.view(
+                    -1, padded_n // block_n, k // vnni_size, vnni_size, block_n
+                )
+                .transpose(-1, -2)
+                .contiguous()
+                .view(-1, padded_n // block_n, k, block_n)
+            )
+            # normalize stride to be "contiguous_strides" per size
+            # this avoids the problems in L.view during template codegen
+            new_stride = [1]
+            for sz in reversed(blocked_w.shape[1:]):
+                new_stride.insert(0, new_stride[0] * sz)
+            blocked_w = blocked_w.as_strided(blocked_w.shape, new_stride)
+        new_inputs[1] = blocked_w
+        return new_inputs, layout_or_out
+
+    def _get_default_reindexers(self, epilogue_nodes):
```

Comment:
This is an override? Suggest to make the name public: `get_default_reindexers` if so.